### DISCUSSION TITLE
RemixI18Next.getRouteNamespaces returns every namespace?

### DISCUSSION THREAD

**Discussion by jrestall** [sergiodxa/remix-i18next#208] [Q&A] [answered]
The docs for [getRouteNamespaces](https://github.com/sergiodxa/remix-i18next/blob/73d3765756f4f4fc5515a0222f517be547825a2a/src/server.ts#L128) imply that it only returns the namespaces for the current request, however I'm finding it returns every namespace in my app.

This causes the server to load a lot more namespaces from an http backend than necessary to serve the current request. Is this done on purpose or could I change it to use just the current route `matches` instead of all `routeModules` instead?

e.g.
```ts
function getRouteNamespaces(context: EntryContext): string[] {
  const namespaces = Object.values(
    context.staticHandlerContext.matches
  ).flatMap((match) => {
    const route = match.route;

    if (typeof route?.handle !== "object") return [];
    if (!route.handle) return [];
    if (!("i18n" in route.handle)) return [];
    if (typeof route.handle.i18n === "string") return [route.handle.i18n];

    if (
      Array.isArray(route.handle.i18n) &&
      route.handle.i18n.every((value: string) => typeof value === "string")
    ) {
      return route.handle.i18n as string[];
    }

    return [];
  });

  return [...new Set(namespaces)];
}
```

**Comment by sergiodxa** [comment, marked as answer]
This should be fixed in v7 as RR only sends the manifest of the matching routes during SSR and not every route
